addons/mail/static/src/web/messaging_menu/messaging_menu.xml

```diff
@@ -77,7 +77,7 @@
             <t t-foreach="displayedPreviews" t-as="preview" t-key="preview.id">
                 <NotificationItem
                     body="preview.message?.inlineBody"
-                    count="preview.count"
+                    counter="preview.isNeedaction ? preview.counter : preview.counter > 1 ? preview.counter : undefined"
```

Comment:
I feel like it is better to not introduce `undefined` here. Could just use 0 for that, so that later it is more safe to make the comparison without being annoyed by undefined. And also, 0 preview also makes sense.

Reply:
`0` and `undefined` does not mean the same.
- `undefined` means no counter
- `0` means show counter of `0`

Reply:
Yes, that's why I think we should make it consistent. `counter` should be there even if the count goes to 0. Imagine we have a counter in real life and count from 5, 4, 3, 2, 1, and when going to 0. We don't say the counter disappears, we say the counter counts to 0. Besides, the `undefined` technically could be more possible to produce a bug.

Reply:
If we don't want to display the 0 we don't need to use `undefined`, we could only show the counter if it's greater than 0, directly in the NotificationItem template. It feel better because it's the component it self that decide and not his parent who should only passe a number.

Reply:
Actually it's not `0` that is problematic, but `1`:
- We want to display counter in failure when strictly above 1 (2+)
- We want to display counter in needaction when strictly above 0 (1+)

So we can't just let `<Notificationitem>` decide display of counter on a threshold when it depends on the type of the notification item. That's why I made it so that `undefined` means display no counter: this lets the user of `<NotificationItem>` decide whether it should display the counter or not.